File: ABC433/abc433_c.py
# ABC433 C - 1122 Substring 2
# See problem: https://atcoder.jp/contests/abc433/tasks/abc433_c
# 2025-11-22 WA->AC+

import sys

def main():
    s = sys.stdin.buffer.read().decode().strip()
    if not s:
        print(0)
        return

    ans = 0

    prev_digit = None
    prev_len = 0

    curr_digit = s[0]
    curr_len = 1

    for ch in s[1:]:
        if ch == curr_digit:
            curr_len += 1
        else:
            if prev_digit is not None:
                if (ord(curr_digit) - ord('0')) == (ord(prev_digit) - ord('0')) + 1:
                    ans += min(prev_len, curr_len)

            prev_digit = curr_digit
            prev_len = curr_len
            curr_digit = ch
            curr_len = 1

    if prev_digit is not None:
        if (ord(curr_digit) - ord('0')) == (ord(prev_digit) - ord('0')) + 1:
            ans += min(prev_len, curr_len)

    print(ans)

if __name__ == "__main__":
    main()

File: ABC433/abc433_c.R
# ABC433 C - 1122 Substring 2
# See problem: https://atcoder.jp/contests/abc433/tasks/abc433_c
# 2025-11-22 WA->AC-

S <- trimws(readLines("stdin", n = 1))

digits <- as.integer(strsplit(S, "", fixed = TRUE)[[1]])
n <- length(digits)

ans <- 0L

prev_num <- NA_integer_
prev_len <- 0L
curr_num <- digits[1]
curr_len <- 1L

if (n >= 2) {
  for (i in 2:n) {
    if (digits[i] == curr_num) {
      curr_len <- curr_len + 1L
    } else {
      if (!is.na(prev_num) && curr_num == prev_num + 1L) {
        ans <- ans + min(prev_len, curr_len)
      }
      prev_num <- curr_num
      prev_len <- curr_len
      curr_num <- digits[i]
      curr_len <- 1L
    }
  }
}

if (!is.na(prev_num) && curr_num == prev_num + 1L) {
  ans <- ans + min(prev_len, curr_len)
}

cat(ans, "\n")
